classes/Search.php

```diff
@@ -137,7 +137,23 @@ class SearchCore
 
     public static function extractKeyWords($string, $id_lang, $indexation = false, $iso_code = false)
     {
-        $sanitizedString = Search::sanitize($string, $id_lang, $indexation, $iso_code, false);
+/*         start improvement
+         it allowss the '&' to be indexed for product names like J&B*/
+        if (!stripos($string, '&')) {
+            $sanitizedString = Search::sanitize($string, $id_lang, false, Context::getContext()->language->iso_code);
```

Comment:
Before, code was
```
$sanitizedString = Search::sanitize($string, $id_lang, $indexation, $iso_code, false);
```
You changed it to
```
$sanitizedString = Search::sanitize($string, $id_lang, false, Context::getContext()->language->iso_code);
```
So you force `$indexation` to false. Why?

Reply:
Hi,
I didn't change/added the false flag, my changed are only the ones included in the /*start improvement*/,  /*end improvement*/